corehq/apps/app_manager/util.py

```diff
@@ -658,3 +658,20 @@ def get_info(self):
             "latest_apk_version": self.get_latest_apk_version(),
             "latest_ccz_version": self.get_latest_app_version(),
         }
+
+
+def get_form_source_download_url(xform):
+    """Returns the download url for the form source for a submitted XForm
+    """
+    from corehq.apps.app_manager.models import Application
+    app = Application.get(xform.build_id)
```

Comment:
Will this succeed if the application has been deleted?

Reply:
yup

Reply:
Also why is this build_id instead of app_id? I just looked at a random form submission and saw `build_id` was null, but it's an old and local form so I don't know how representative that is.

Reply:
hmm, I do see now that the other link references `build_id` so maybe my form is an anomaly

Reply:
part of the point of this is that you get the source for the build that this XForm was submitted against. I thought `build_id` referred to that?